src/user/user.service.spec.ts

```diff
@@ -1,36 +1,78 @@
-import { UserModule } from "./user.module";
+
 import { Test, TestingModule } from "@nestjs/testing";
-import { UserService } from "./user.service";
-import { expect } from "chai";
-import { DatabaseModule } from "../database/database.module";
-import { userProviders } from "./user.providers";
 import { ConfigModule } from "@nestjs/config";
 import { IUserService } from "./user.types";
+import { ProviderTokens } from "../tokens";
+import { Repository } from "typeorm";
+import { User } from "../database/entities/user.entity";
+import { UserService } from "./user.service";
+
+const userRepositoryMock = {
+    findOne: (): Promise<User> => {
+        return new Promise((resolve, reject) => {
+            const result: User = {
+                id: "1",
+                email: "test@test.com",
+                createdAt: new Date(),
+                updatedAt: new Date(),
+            }
+            resolve(result);   
+        }); 
+    },
+    find: (): Promise<User[]> => {
+        return new Promise((resolve, reject) => {
+            const result: User[] =[
+                {
+                    id: "1",
+                    email: "test@test.com",
+                    createdAt: new Date(),
+                    updatedAt: new Date(),
+                },
+                {
+                    id: "2",
+                    email: "test2@test.com",
+                    createdAt: new Date(),
+                    updatedAt: new Date(),
+                }
+            ]
+            resolve(result);
+        });
+    }
+};
```

Comment:
Nice! Exactly the sort of thing I was thinking of.

Reply:
Oh - I've just noticed that it's now dying when it fails to find the DB :( 

This sort of thing is where we can provide real value to new projects, but If you find yourself getting bogged down here, do grab me - I should be able recognise at least some of the problems 🤞